test/tools/libtesteth/wast2wasm.cpp

```diff
@@ -0,0 +1,55 @@
+#include "wast2wasm.h"
+
+#include <libdevcore/CommonData.h>
+
+#include <boost/test/unit_test.hpp>
+
+// Disable MSVC warning "unary minus operator applied to unsigned type".
+#pragma warning(push)
+#pragma warning(disable: 4146)
+#include <wasm-binary.h>
+#include <wasm-s-parser.h>
+#include <wasm-validator.h>
+#pragma warning(pop)
+
+using namespace std;
+
+namespace dev
+{
+
+namespace test
+{
+
+/* from https://github.com/ewasm/evm2wasm */
+string wast2wasm(string const& input, bool debug)
+{
+    wasm::Module module;
+
+    try {
+        // FIXME: binaryen 1.37.28 actually modifies the input...
+        //        as a workaround make a copy here
+        string tmp = input;
+        wasm::SExpressionParser parser(const_cast<char*>(tmp.c_str()));
```

Comment:
You don't need the cast now.

Reply:
Updated.

Reply:
A cast is still needed as `c_str()` returns `const char*` and `SExpressionParser` wants `char*`. Do you suggest any other cast / solution?